widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java

```diff
@@ -187,71 +187,219 @@ private boolean isPermissionDataMediaRequestCode(Pair<Integer, Integer[]> permis
+        newIntent.putExtra(GliaWidgets.USE_OVERLAY, configuration.sdkConfiguration.getUseOverlay());
+        newIntent.putExtra(GliaWidgets.SCREEN_SHARING_MODE, configuration.sdkConfiguration.getScreenSharingMode());
         startActivity(newIntent);
     }
 
     private void navigateToSurvey(Survey survey) {
         Intent newIntent = new Intent(getApplicationContext(), SurveyActivity.class);
-        newIntent.putExtra(GliaWidgets.UI_THEME, configuration.getRunTimeTheme());
+        newIntent.putExtra(GliaWidgets.UI_THEME, configuration.sdkConfiguration.getRunTimeTheme());
         newIntent.putExtra(GliaWidgets.SURVEY, (Parcelable) survey);
         startActivity(newIntent);
     }
 
-    // TODO create builder class
+    @Deprecated
     public static Intent getIntent(
             Context applicationContext,
             GliaSdkConfiguration sdkConfiguration,
             String mediaType
     ) {
-        return getIntent(applicationContext, sdkConfiguration, mediaType, false);
+        return getIntent(applicationContext,
+                Configuration.Builder
+                        .builder()
+                        .setWidgetsConfiguration(sdkConfiguration)
+                        .setMediaType(toMediaType(mediaType))
+                        .build()
+        );
     }
 
     public static Intent getIntent(
-            Context applicationContext,
-            GliaSdkConfiguration sdkConfiguration,
-            String mediaType,
-            boolean isUpgradeToCall
+            Context context,
+            CallActivity.Configuration configuration
     ) {
-        Intent intent = new Intent(applicationContext, CallActivity.class);
-        intent.putExtra(GliaWidgets.COMPANY_NAME, sdkConfiguration.getCompanyName());
-        intent.putExtra(GliaWidgets.QUEUE_ID, sdkConfiguration.getQueueId());
-        intent.putExtra(GliaWidgets.CONTEXT_URL, sdkConfiguration.getContextUrl());
-        intent.putExtra(GliaWidgets.UI_THEME, sdkConfiguration.getRunTimeTheme());
-        intent.putExtra(GliaWidgets.USE_OVERLAY, sdkConfiguration.getUseOverlay());
-        intent.putExtra(GliaWidgets.SCREEN_SHARING_MODE, sdkConfiguration.getScreenSharingMode());
-        intent.putExtra(GliaWidgets.MEDIA_TYPE, mediaType);
-        intent.putExtra(GliaWidgets.IS_UPGRADE_TO_CALL, isUpgradeToCall);
-        return intent;
+        return IntentBuilder
+                .from(context)
+                .setConfiguration(configuration)
+                .getIntent();
+    }
+
+    public static Engagement.MediaType toMediaType(String mediaType) {
+        switch (mediaType) {
+            case GliaWidgets.MEDIA_TYPE_VIDEO:
+                return Engagement.MediaType.VIDEO;
+            case GliaWidgets.MEDIA_TYPE_AUDIO:
+                return Engagement.MediaType.AUDIO;
+            default:
+                throw new InvalidParameterException("Invalid Media Type");
+        }
+    }
+
+    public static class Configuration {
```

Comment:
does the Configuration and all other static builder classes need to be inner classes? They should not bloat the CallActivity.java.